addons/udes_stock/models/stock_move_line.py

```diff
@@ -332,6 +332,84 @@ def prepare(
                 quantity_fulfilled = 0
                 if is_serial:
                     qty_done = 1
+
+                # Swapping the tracked products
+                lot_names_on_picking = self.lot_id.mapped("name")
+                lot_names_swapped_in = list(set(lot_names) - set(lot_names_on_picking))
+                # If there swapping tracked products are allowed and some lot_names have been swapped in
+                if mls.u_picking_type_id.u_allow_swapping_tracked_products and lot_names_swapped_in:
```

Comment:
This would not raise an error currently, I think it probably should. This method (as far as I can see) only gets called in the flow, so coming from the flow we could never have that situation, but if it was used elsewhere I guess we could